src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash3.cs

```diff
@@ -0,0 +1,884 @@
+            DeriveSecretFromSeed(secret, 0);
+            Debug.Assert(new Span<byte>(secret, SecretLengthBytes).SequenceEqual(DefaultSecret));
+
+            // Validate some relationships.
+            Debug.Assert(InternalBufferLengthBytes % StripeLengthBytes == 0);
+        }
+#endif
+
+        private State _state;
+
+        /// <summary>Initializes a new instance of the XXH3 class using the default seed value 0.</summary>
+        public XxHash3() : this(0)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the XXH3 class using the specified seed.</summary>
+        public XxHash3(long seed) : base(HashLengthInBytes)
+        {
+            _state.Seed = (ulong)seed;
+
+            fixed (byte* secret = _state.Secret)
+            {
+                if (seed == 0)
+                {
+                    DefaultSecret.CopyTo(new Span<byte>(secret, SecretLengthBytes));
+                }
+                else
+                {
+                    DeriveSecretFromSeed(secret, (ulong)seed);
+                }
+            }
+
+            Reset();
+        }
+
+        /// <summary>Computes the XXH3 hash of the provided data.</summary>
+        /// <param name="source">The data to hash.</param>
+        /// <returns>The XXH3 64-bit hash code of the provided data, written in little-endian order.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        public static byte[] Hash(byte[] source) => Hash(source, seed: 0);
+
+        /// <summary>Computes the XXH3 hash of the provided data using the provided seed.</summary>
+        /// <param name="source">The data to hash.</param>
+        /// <param name="seed">The seed value for this hash computation.</param>
+        /// <returns>The XXH3 64-bit hash code of the provided data, written in little-endian order.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        public static byte[] Hash(byte[] source, long seed)
+        {
+#if NET6_0_OR_GREATER
+            ArgumentNullException.ThrowIfNull(source);
+#else
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+#endif
+
+            return Hash(new ReadOnlySpan<byte>(source), seed);
+        }
+
+        /// <summary>Computes the XXH3 hash of the provided data using the optionally provided seed.</summary>
+        /// <param name="source">The data to hash.</param>
+        /// <param name="seed">The seed value for this hash computation. The default is zero.</param>
+        /// <returns>The XXH3 64-bit hash code of the provided data, written in little-endian order.</returns>
+        public static byte[] Hash(ReadOnlySpan<byte> source, long seed = 0)
+        {
+            byte[] result = new byte[HashLengthInBytes];
+            BinaryPrimitives.WriteInt64LittleEndian(result, HashToInt64(source, seed));
```

Comment:
The spec for XxHash32/XxHash64 says the canonical order for the output of those algorithms is big endian.  Why have you chosen little endian here, is there precedent from the reference implementation for that?

https://github.com/Cyan4973/xxHash/blob/dev/doc/xxhash_spec.md#step-7-output

> Step 7. Output
>
> The XXH32() function produces an unsigned 32-bit value as output.
> 
> For systems which require to store and/or display the result in binary or hexadecimal format, the canonical format is defined to reproduce the same value as the natural decimal format, hence follows **big-endian** convention (most significant byte first).

<sub>emphasis in the original</sub>

Reply:
> is there precedent from the reference implementation for that?

The reference implementation doesn't produce bytes, it produces numbers.  We're the only implementation I can find that insists on producing bytes from these xxHash algorithms.

I'll change it to use big-endian.  It's just unfortunate that 99.9999% of systems in the world have to expend any additional energy to do so.